compiler/resolution/lowerIterators.cpp

```diff
@@ -1528,7 +1553,7 @@ getIteratorChildren(Vec<Type*>& children, Type* type) {
 
 static void
 buildIteratorCallInner(BlockStmt* block, Symbol* ret, int fnid, Symbol* iterator) {
-  IteratorInfo* ii = iterator->type->defaultInitializer->getFormal(1)->type->defaultInitializer->iteratorInfo;
+  IteratorInfo* ii = getTheIteratorFn(iterator)->iteratorInfo;
```

Comment:
Also, explaining the relationship between "TheIteratorFn" and the IteratorInfo would be good.  If the contents of IteratorInfo and how the functions listed there are derived from TheIteratorFn is documented elsewhere, a pointer to that documentation would also be helpful to the uninitiated.